### ISSUE TITLE
Track immediate `Sources` when possible

### ISSUE THREAD

**Issue by tsalo** [nipreps/fmriprep#3261]
### What would you like to see added in fMRIPrep?

1. Standard-space and T1w-space outputs do not include `Sources`.
    - The `Sources` should probably be the boldref-space outputs, when those are requested. Otherwise, they should include the original data, the HMC transforms, the coreg transforms, the ref2anat transform, and the anat2std transform.
2. Echo-wise outputs do not include `Sources`.
3. Surface files don't seem to include `Sources`.

### Do you have any interest in helping implement the feature?

Yes

**Comment by tsalo** [comment]
@effigies do you think it would be worth it to try building the TemplateFlow templates into the DatasetLinks and Sources?

**Comment by effigies** [comment]
Yes, I think including templateflow would be very helpful both for sources and spatial references.

**Comment by tsalo** [comment]
Want to note that `from-orig_to-boldref_mode-image_desc-hmc_xfm.txt` isn't in the `Sources`.

`from-boldref_to-auto00000_mode-image_xfm.txt` isn't either, but I can't tell if that's accounted for by including `from-boldref_to-T1w_mode-image_xfm.txt` in the Sources. @effigies do you know? If `orig-to-boldref`, `boldref-to-fmap` and `boldref-to-T1w` are _all_ strung together, then I think `boldref-to-T1w` might be mislabeled.

**Comment by effigies** [comment]
Each of these is an atomic transform. orig-to-boldref + boldref-T1w are strung together to get the individual volume there, but boldref-to-fmap is used to transform the fieldmap into the boldref space and calculate a voxel-shift-map. We do not save the VSM, so it would be appropriate to include both the fieldmap and the transform as inputs. The exact provenance would be better modeled using `prov.jsonld`.

**Comment by tsalo** [comment]
Ah, thanks! I think SDCFlows will need to pass its fmap file from its derivatives workflow in order to use it as a source in fMRIPrep.

**Comment by effigies** [comment]
I'd forgotten about this and opened #3534. I made it a sub-issue so it's linked prominently.
